hbase-server/src/main/java/org/apache/hadoop/hbase/master/normalizer/RegionNormalizerWorker.java

```diff
@@ -184,18 +195,22 @@ public void run() {
     while (true) {
       if (Thread.interrupted()) {
         LOG.debug("interrupt detected. terminating.");
+        pool.shutdown();
         break;
       }
       final TableName tableName;
       try {
         tableName = workQueue.take();
       } catch (InterruptedException e) {
         LOG.debug("interrupt detected. terminating.");
+        pool.shutdown();
         break;
       }
 
-      final List<NormalizationPlan> plans = calculatePlans(tableName);
-      submitPlans(plans);
+      pool.submit(() -> {
+        final List<NormalizationPlan> plans = calculatePlans(tableName);
+        submitPlans(plans);
```

Comment:
The name here is a bit confusing, if it just 'submits' the plan, it should not take a lot of time? Does it also execute the plan?